pennylane/templates/state_preparations/state_prep_mps.py

```diff
@@ -15,24 +15,234 @@
+    assert qml.math.log2(
+        dj2
+    ).is_integer(), "The second dimension of the first tensor must be a power of 2."
+
+    # Validate the shapes of the intermediate tensors
+    for i, array in enumerate(mps[1:-1], start=1):
+        shape = qml.math.shape(array)
+        assert qml.math.isclose(len(shape), 3), f"Tensor {i} must have exactly 3 dimensions."
+        new_dj0, new_dj1, new_dj2 = shape
+        assert qml.math.isclose(
+            new_dj1, 2
+        ), f"The second dimension of tensor {i} must be exactly 2."
+        assert qml.math.log2(
+            new_dj0
+        ).is_integer(), f"The first dimension of tensor {i} must be a power of 2."
+        assert qml.math.isclose(
+            new_dj1, 2
+        ), f"The second dimension of tensor {i} must be exactly 2."
+        assert qml.math.log2(
+            new_dj2
+        ).is_integer(), f"The third dimension of tensor {i} must be a power of 2."
+        assert qml.math.isclose(
+            new_dj0, dj2
+        ), f"Dimension mismatch: tensor {i}'s first dimension does not match the previous third dimension."
+        dj2 = new_dj2
+
+    # Validate the shape and dimensions of the last tensor
+    assert qml.math.isclose(
+        len(qml.math.shape(mps[-1])), 2
+    ), "The last tensor must have exactly 2 dimensions."
+    new_dj0, new_dj1 = qml.math.shape(mps[-1])
+    assert new_dj1 == 2, "The second dimension of the last tensor must be exactly 2."
+    assert qml.math.log2(
+        new_dj0
+    ).is_integer(), "The first dimension of the last tensor must be a power of 2."
+    assert qml.math.isclose(
+        new_dj0, dj2
+    ), "Dimension mismatch: the last tensor's first dimension does not match the previous third dimension."
+
+
+def right_canonicalize_mps(mps):
+    r"""Transform a matrix product state (MPS) into its right-canonical form.
+
+    Args:
+        mps (list[TensorLike]): List of tensors representing the MPS.
+
+    Returns:
+        List of tensors representing the MPS in right-canonical form.
+
+
+    A right-canonicalized MPS is a matrix product state where each tensor :math:`A^{(j)}` satisfies
+    the following orthonormality condition:
+
+    .. math::
+
+        \sum_{d_{j,1}, d_{j,2}} A^{(j)}_{d_{j, 0}, d_{j, 1}, d_{j, 2}} \left( A^{(j)}_{d'_{j, 0}, d_{j, 1}, d_{j, 2}} \right)^* = \delta_{d_{j, 0}, d'_{j, 0}}
+
+    where :math:`d_{i,j}` denotes the :math:`j` dimension of the :math:`i` tensor And :math:`\delta` is a
+    function that takes the value :math:`1` if the two inputs are the same and :math:`0` otherwise.
+    The definition can be found in Eq. (21) of `arXiv:2310.18410 <https://arxiv.org/pdf/2310.18410>`_.
+
+    .. seealso:: :class:`~.MPSPrep`.
+
+    **Example**
+
+    .. code-block::
+
+        n_sites = 4
+
+        mps = ([np.ones((2, 4))] +
+               [np.ones((4, 2, 4)) for _ in range(1, n_sites - 1)] +
+               [np.ones((4, 2))])
+
+        mps_rc = qml.right_canonicalize_mps(mps)
```

Comment:
Can we then instead write an assert statement for showing it is actually right canonicalized?  Use may use `tensordot` or `einsum` to make it compact.